### PR TITLE
don't use TLS when being redirected from http to https

### PR DESCRIPTION
**Pull request by MartinNowak**
- disconnect when changing host, schema, or port
- add defaultPort to URL to simplify code

### CODE REVIEW

File: source/vibe/http/client.d [hunk 1]
```diff
@@ -244,6 +244,7 @@ final class HTTPClient {
 		HTTPClientSettings m_settings;
 		string m_server;
 		ushort m_port;
+		bool use_tls;
```
Comment (s-ludwig) [inline source/vibe/http/client.d]: Should be named `m_useTLS`.

**Comment by s-ludwig** [comment]
Apart from the naming and the merge conflict in client.d, LGTM!

**Comment by s-ludwig** [comment]
I was wondering if it would make sense to make the `.port` property of `URL` smart, so that it automatically uses the default port when no explicit port is given (and sets the port to 0 if it equals `defaultPort`). Maybe with an additional `rawPort` property that accesses the actual port written in the URL string representation.

**Comment by MartinNowak** [comment]
Updated

> I was wondering if it would make sense to make the .port property of URL smart, so that it automatically uses the default port when no explicit port is given

I think `url.port ? url.port : url.defaultPort` is short enough and less complex.

**Comment by MartinNowak** [comment]
Ping @s-ludwig.

**Comment by s-ludwig** [comment]
LGTM!

**Pull request merged**